src/macro.jl

```diff
@@ -79,12 +79,18 @@ ERROR: AssertionError: Float64($(Expr(:escape, :sqrt5))) == Float64(big($(Expr(:
 ```
 """
 macro irrational(sym::Symbol, val::Float64, def::Union{Symbol,Expr}, T::Symbol=Symbol(uppercasefirst(string(sym))))
-    irrational(sym, val, def, T)
+    irrational(__module__, sym, val, def, T)
 end
 macro irrational(sym::Symbol, def::Union{Symbol,Expr}, T::Symbol=Symbol(uppercasefirst(string(sym))))
-    irrational(sym, :(big($(esc(sym)))), def, T)
+    irrational(__module__, sym, :(big($(esc(sym)))), def, T)
 end
-function irrational(sym::Symbol, val::Union{Float64,Expr}, def::Union{Symbol,Expr}, T::Symbol)
+function irrational(mod::Module, sym::Symbol, val::Union{Float64,Expr}, def::Union{Symbol,Expr}, T::Symbol)
+    if isdefined(mod, T)
+        throw(ArgumentError("Type `$T` is already defined in module `$mod`."))
```

Comment:
IMO the error message should be more descriptive, e.g.
```suggestion
        throw(ArgumentError("Type `$T` of irrational constant `$sym` is already defined in module `$mod`."))
```